pkg/signup/signup_service_test.go

```diff
@@ -403,6 +437,89 @@ func (s *TestSignupServiceSuite) TestGetSignupUnknownStatus() {
 	require.EqualError(s.T(), err, "unable to parse readiness status as bool: blah-blah-blah: strconv.ParseBool: parsing \"blah-blah-blah\": invalid syntax")
 }
 
+func (s *TestSignupServiceSuite) TestGetUserSignup() {
+	svc, fakeClient, _, _ := newSignupServiceWithFakeClient()
+
+	s.Run("getusersignup ok", func() {
+		us := s.newUserSignupComplete()
+		err := fakeClient.Tracker.Add(us)
+		require.NoError(s.T(), err)
+
+		val, err := svc.GetUserSignup(us.Name)
+		require.NoError(s.T(), err)
+		require.Equal(s.T(), us.Name, val.Name)
+	})
+
+	s.Run("getusersignup returns error", func() {
+		fakeClient.MockGet = func(s string) (userSignup *v1alpha1.UserSignup, e error) {
+			return nil, errors.New("get failed")
+		}
+
+		val, err := svc.GetUserSignup("foo")
+		require.Error(s.T(), err)
+		require.Nil(s.T(), val)
+	})
+
+	s.Run("getusersignup returns not found error", func() {
```

Comment:
Well... technically it doesn't return an error. Which is expected:)

Reply:
I've modified the description.